proposals/repo-to-tar/README.md

```diff
@@ -0,0 +1,106 @@
+
+`imgpkg push -b index.docker.io/user1/simple-app-bundle:v1.0.0 -f examples/basic-step-2` <br>
+`imgpkg copy -b index.docker.io/user1/simple-app-bundle:v1.0.0 --to-tar /tmp/my-image.tar`
+
+This has 3 fold issues : <br>
+1. Dependent on using a registry to create a tar file from a bundle image.
+2. The tar file is created from the bundle image, which is not required if the user just wants to create a tar file having a layer as tar with configurations/metadata.
+3. The bundle or image and it's tar file formed is not OCI compliant. This proposal aims to make the tar in a oci spec compliant format.
+
+## Terminology / Concepts
+1. Definitions of terms used in the proposal with respect to imgpkg can be found [here.](https://carvel.dev/imgpkg/docs/v0.37.x/)<br>
+2. Resources regarding docker save and docker load can be found [here.](https://docs.docker.com/engine/reference/commandline/save/)<br>
+3. A good article to understand OCI Artifacts can be found [here.](https://dlorenc.medium.com/oci-artifacts-explained-8f4a77945c13)
+4. Read about oci-spec in detail [here.](https://github.com/opencontainers/image-spec/blob/main/image-layout.md)
+
+## Proposal
+
+#### imgpkg
+
+Oci Tar Creation : <br>
+
+imgpkg save command will be added to imgpkg. This command will create a tar file from the bundle image. The command will be used as follows : <br>
+`imgpkg save -f some-folder --to-oci-tar local-oci-format.tar`<br>
+It should generate and save tar file at /tmp/my-image.
+
+And also to make it compatible to do both, pushing folders as bundle directly to the registry whilist also obtaining a tar image, we will similarily add a --oci-tar flag : <br>
+`imgpkg push -b registry.example.com/xyz -f some-folder --to-oci-tar local-oci-format.tar`
+
+Oci Tar Pushing : <br>
+
+Now in order to push the tar file to the registry, we can use the already existing copy command with a new flag as follows : <br>
+`imgpkg copy --oci-tar local-oci-format.tar --to-repo registry.example.com/abc`
+
+### Goals and Non-goals and Future-goals
+
+#### Goals
+1. Making OCI Complaint tar and image : <br>
+Create and save the image in oci format locally along with tar creation.
+An example of directory structure formed from complying to [image spec](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) : <br>
+
+```md
+├── blobs
+│   └── sha256
+│       ├── 45e16698e6e721de091b5ecbb811eae1243fb8f2ed5dbd04a9f40cd199c355f9
+│       ├── 4da097607a3cb58407cd8c157271bb5f7462a5872d516a5bc979ec678ac30f7d
+│       └── 5c2dafe3c70c13990190d643c91e9f67b8129b179257674888178868474f6511
+├── index.json
+└── oci-layout
+```
+where one of the files in sha256 is the actual file tar and others 2 are configs and manifests.<br>
+
+2. Add functionality to `imgpkg save` to create and persist a oci tar file mentioned in point 1 which in turn is compatible with `imgpkg copy` command.<br>
```

Comment:
As I said above I think we can skip this goal

Reply:
Removed the save command from the proposal.